Plandomizer.py

```diff
@@ -970,20 +970,26 @@ def __init__(self, settings, src_dict=None):
         self.settings.__dict__.update(update_dict['_settings'])
         if 'settings' in self.src_dict:
             for setting, choice in self.src_dict['settings'].items():
-                if not is_mapped(setting):
+                # Ensure the supplied setting name is a real setting
+                if setting not in [x.name for x in setting_infos]:
                     raise TypeError('%s is not a valid setting. %s' % (repr(setting), build_close_match(setting, 'setting')))
                 info = get_setting_info(setting)
+                # Ensure the type of the supplied choice is correct
                 if type(choice) != info.type:
                     raise TypeError('Supplied choice %s for setting %s is of type %s, expecting %s' % (repr(choice), repr(setting), repr(type(choice).__name__), repr(info.type.__name__)))
-                if not isinstance(choice, list) and choice not in info.choice_list:
-                    if setting == 'compress_rom' and choice == 'Temp':
-                        continue
-                    raise ValueError('%s is not a valid choice for setting %s. %s' % (repr(choice), repr(setting), build_close_match(choice, 'choice', info.choice_list)))
                 # If setting is a list, must check each element
-                elif isinstance(choice, list):
+                if isinstance(choice, list):
                     for element in choice:
                         if element not in info.choice_list:
-                            raise ValueError('%s is not a valid choice for setting %s. %s' % (repr(element), repr(setting), build_close_match(choice, 'choice', info.choice_list)))
+                            raise ValueError('%s is not a valid choice for setting %s. %s' % (repr(element), repr(setting), build_close_match(element, 'choice', info.choice_list)))
```

Comment:
I should have noticed this before--you can just use `'%r' % element` instead of `'%s' % repr(element)`

Reply:
Thanks for the heads up, that's way cleaner!